### ISSUE TITLE
Duplicate call of self.lobby_addr.do_send(Disconnect { id: self.id, room_id: self.room });

### ISSUE THREAD

**Issue by tadeohepperle** [antholeole/actix-sockets#6]
Hey, I think there is a bug in this section of the code:
```rs
fn hb(&self, ctx: &mut ws::WebsocketContext<Self>) {
        ctx.run_interval(HEARTBEAT_INTERVAL, |act, ctx| {
            if Instant::now().duration_since(act.hb) > CLIENT_TIMEOUT {
                println!("Disconnecting failed heartbeat");
                act.lobby_addr.do_send(Disconnect { id: act.id, room_id: act.room });
                ctx.stop();
                return;
            }

            ctx.ping(b"PING");
        });
    }
```
It is unnecessary to call `act.lobby_addr.do_send(Disconnect { id: act.id, room_id: act.room });` here, because the `ctx.stop();` will cause the actor to enter the stopping state. In that case the body of the `stopping()` function is run, which already sends a Disconnect message: 

```rs
fn stopping(&mut self, _: &mut Self::Context) -> Running {
        self.lobby_addr.do_send(Disconnect { id: self.id, room_id: self.room });
        Running::Stop
    }
```
If you give me permissions to make a PR I'll fix it quickly

**Comment by antholeole** [comment]
Yep, just tested and this is correct! Thanks for finding this. I'll accept a PR if you create one :).

**Comment by tadeohepperle** [comment]
I have a PR ready, but when I push it it says: "remote: Permission to antholeole/actix-sockets.git denied to tadeohepperle."

**Comment by antholeole** [comment]
You have to fork the repository, commit the change to your fork, and then open a PR through the GitHub UI :). 

Alternatively, you could just open up the file in GitHub and do it through GitHub’s editor, since it’s not a large change.

If you don’t care about being marked as a contributor I can do it myself!

**Comment by antholeole** [comment]
closed with #7 - thanks!
